src/utils/datetools.ts

```diff
@@ -1,3 +1,37 @@
+const EXPECTED_LOCALE = 'it-IT';
+
+const options: Record<string, Intl.DateTimeFormatOptions> = {
+  romeDate: {
+    timeZone: 'Europe/Rome',
+    day: '2-digit',
+    month: '2-digit',
+    year: 'numeric'
+  }
+};
+
+const romeDateIntlIt = new Intl.DateTimeFormat('it-It', options.romeDate);
+
+const isoToHumanDateRome = (isoDateString?: string, locale = navigator.language) => {
+  if (!isoDateString) {
+    return '';
+  }
+
+  const date = new Date(isoDateString);
+
+  // Check if the date is invalid
+  if (isNaN(date.getTime())) {
+    return '';
+  }
+
+  // Use Intl instance for performance optimization
+  // on the expected most used locale
+  if (locale === EXPECTED_LOCALE) {
+    return romeDateIntlIt.format(date);
+  } else {
+    return date.toLocaleString(locale, options.romeDate);
```

Comment:
Could you add a test with a locale different from the EXPECTED LOCALE to have this branch covered?

Reply:
I'm going to remove the optimization code